### PR TITLE
Resolve dependencies before filling sections

### PR DESCRIPTION
**Pull request by sils1297**
Fixes https://github.com/coala-analyzer/coala/issues/2709

### CODE REVIEW

**Comment by chauffer** [comment]
Does this need tests? write a test man

**Comment by sils1297** [comment]
:+1:

File: coalib/output/printers/LogPrinter.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class LogPrinter:
 
     def __init__(self,
                  printer,
-                 log_level=LOG_LEVEL.INFO,
+                 log_level=LOG_LEVEL.DEBUG,
```
Comment (chauffer) [inline coalib/output/printers/LogPrinter.py]: Why is this here?

**Comment by Asnelchristian** [comment]
ack https://github.com/coala-analyzer/coala/pull/2717/commits/198b09240221e136d6513d2e73f70a28e4e67714

**Comment by Asnelchristian** [comment]
ack https://github.com/coala-analyzer/coala/pull/2717/commits/688579b29e5e5a2a1fa9e33d6ce67b625111be8b

**Comment by sils1297** [comment]
@rultor merge
